lib/puppet_x/puppetlabs/powershell/powershell_manager.rb

```diff
@@ -33,18 +31,34 @@ def self.supported?
         Puppet::Util::Platform.windows? && !win32console_enabled?
       end
 
-      def initialize(cmd)
+      def initialize(cmd, debug)
         @usable = true
+
+        named_pipe_name = "#{SecureRandom.uuid}PuppetPsHost"
+
+        ps_args = ['-File', self.class.init_path, "\"#{named_pipe_name}\""]
+        ps_args << '"-EmitDebugOutput"' if debug
         # @stderr should never be written to as PowerShell host redirects output
-        @stdin, @stdout, @stderr, @ps_process = Open3.popen3(cmd)
+        stdin, @stdout, @stderr, @ps_process = Open3.popen3("#{cmd} #{ps_args.join(' ')}")
+        stdin.close
 
         Puppet.debug "#{Time.now} #{cmd} is running as pid: #{@ps_process[:pid]}"
 
-        init_ready_event_name =  "Global\\#{SecureRandom.uuid}"
-        init_ready_event = self.class.create_event(init_ready_event_name)
+        pipe_path = "\\\\.\\pipe\\#{named_pipe_name}"
+        # wait for the pipe server to signal ready, and fail if no response in 10 seconds
+
+        # wait up to 10 seconds in 0.2 second intervals to be able to open the pipe
```

Comment:
Is 10 seconds long enough for PowerShell to startup and begin listening?  Or should we wait longer?